### ISSUE TITLE
Submodules vs Git dependencies in requirements.txt

### ISSUE THREAD

**Issue by mryab** [PrimeIntellect-ai/OpenDiloco#1]
Awesome project! I was curious about the decision to use Git submodules instead of directly specifying them as [dependencies in requirements.txt](https://arxiv.org/abs/2407.07852). This should make it a bit easier to install the [Python dependencies](https://github.com/PrimeIntellect-ai/OpenDiloco/blob/main/README.md#environment-setup) you list in README. If you need to install a library with [extra flags](https://pip.pypa.io/en/stable/cli/pip_install/#examples), this should also be possible: https://adamj.eu/tech/2019/03/11/pip-install-from-a-git-repository/

**Comment by samsja** [comment]
Hey @mryab.

The reason was mainly because we started our work from a fork of hivemind. Everything in the `open_diloco/hivemind_diloco.py` was entangle with `hivemind.Optimizer` But since we wanted rather to rely on hivemind as a library we started separate the concerns. So it was either development wise to us to use git submodule. 

But this is legacy indeed, we should swithc to just use the git flag in the requirement.txt :)
